### PR TITLE
3rd party libraries

### PR DESCRIPTION
**Pull request by sgpearse**
To migrate to the new libraries, download the bundle for your OS, linked below.

- On UNIX, extract their contents into /usr/local/VAPOR-Deps
- On Windows, extract the contents into C:\

[Ubuntu](https://drive.google.com/open?id=1j4IO4VCU0Wvyu2T3BH0e9I0qiiwCIrEd)
[OSX](https://drive.google.com/open?id=1kyDAzmK9zShrIcoAbned8YW9vn2j7lpe)
[Win32](https://drive.google.com/open?id=1sRlE06jSVrCScrt546G4UtI3hj93xV2W)
[CentOS](https://drive.google.com/open?id=13H5JwvtBpoN4MQsX3IShbc1tPPaa6o5G)

### CODE REVIEW

File: apps/vaporgui/VolumeIsoSubtabs.cpp [hunk 1]
```diff
@@ -31,22 +31,22 @@ VolumeIsoAppearanceSubtab::VolumeIsoAppearanceSubtab(QWidget* parent)
...
     _diffuseWidget->SetIntType( false );
 
-    _specularWidget->SetLabel( QString::fromAscii("Specular") );
+    _specularWidget->SetLabel( QString::fromLatin1("Specular") );
     _specularWidget->SetDecimals( 2 );
     _specularWidget->SetExtents( 0.0, 1.0 );
     _specularWidget->SetIntType( false );
 
-    _shininessWidget->SetLabel( QString::fromAscii("Shininess") );
+    _shininessWidget->SetLabel( QString::fromLatin1("Shininess") );
```
[the remark this answers is not in the document]
Comment (sgpearse) [reply apps/vaporgui/VolumeIsoSubtabs.cpp]: Good catch.  fromLatin1() is unnecessary.

File: lib/common/ResourcePath.cpp [hunk 2]
```diff
@@ -101,7 +102,7 @@ std::string Wasp::GetSharePath(const std::string &name)
 #ifdef WIN32
 #define PYTHON_INSTALLED_PATH ("python" + string(PYTHON_VERSION))
 #else
-#define PYTHON_INSTALLED_PATH ("lib/python" + string(PYTHON_VERSION))
+#define PYTHON_INSTALLED_PATH ("lib/python3.6")
```
[the remark this answers is not in the document]
Comment (sgpearse) [reply lib/common/ResourcePath.cpp]: Understood.

File: lib/render/MyPython.cpp [hunk 3]
```diff
@@ -82,23 +82,14 @@ int MyPython::Initialize() {
 
 	if (! m_pyHome.empty()) {
 #ifdef WIN32
-		Py_SetPythonHome((char *) m_pyHome.c_str());
+		std::string pythonPath = m_pyHome + "\\Python36;";
+		pythonPath = pythonPath + m_pyHome + "\\Python36\\Lib;";
+		pythonPath = pythonPath + m_pyHome +  "\\Python36\\Lib\\site-packages";
```
[the remark this answers is not in the document]
Comment (sgpearse) [reply lib/render/MyPython.cpp]: We can.  It requires adding a helper function to ResoucePath.

File: apps/vaporgui/TFInfoWidget.cpp [hunk 4]
```diff
@@ -19,8 +19,12 @@ TFInfoWidget::TFInfoWidget()
...
     layout->addWidget(new VLineItem("Data Value", _valueEdit = new QLineEdit));
     _valueEditType = new QComboBox;
-    
-    _valueEdit->setValidator(new QDoubleValidator(__FLT_MIN__, __FLT_MAX__, 7));
+
+#ifdef WIN32
+    _valueEdit->setValidator(new QDoubleValidator(FLT_MIN, FLT_MAX, 7));
+#else
+	_valueEdit->setValidator(new QDoubleValidator(__FLT_MIN__, __FLT_MAX__, 7));
+#endif
```
[the remark this answers is not in the document]
Comment (sgpearse) [reply apps/vaporgui/TFInfoWidget.cpp]: This is an old fix that shouldn't have made it into the PR.

File: apps/vaporgui/CMakeLists.txt [hunk 5]
```diff
@@ -271,8 +272,20 @@ else ()
 	add_executable (vapor ${SRCS} ${UIS})
 endif()
 
-# target_link_libraries (vapor common vdc wasp render params m jpeg dl util GLEW tiff geotiff assimp Qt4::QtCore Qt4::QtGui Qt4::QtOpenGL)
-target_link_libraries (vapor common vdc wasp render params jpeg ${GLEW} tiff geotiff assimp Qt4::QtCore Qt4::QtGui Qt4::QtOpenGL python${PYTHONVERSION})
+if (WIN32)
+	target_link_libraries (
+    		vapor common vdc wasp render params jpeg ${GLEW} 
+    		libtiff geotiff assimp-vc140-mt Qt5::Widgets Qt5::OpenGL Qt5::Core Qt5::Gui Qt5::DBus
```
Comment (StasJ) [inline apps/vaporgui/CMakeLists.txt]: Instead of having all these additional ifdefs, could you just create a variable ASSIMP_LIB and set it to "assimp-vc140-mt" on windows and "assume" everywhere else? (why does it have to be called "assimp-vc140-mt")

File: apps/vaporgui/FileOperationChecker.cpp [hunk 6]
```diff
@@ -92,11 +92,11 @@ FileOperationChecker::FileGoodToWrite( const QString& filename )
     // In case this file does not exist
     if( !fileInfo.exists() )
     {
-        std::FILE* f = std::fopen( filename.toAscii(), "w" );
+        std::FILE* f = std::fopen( filename.toLocal8Bit(), "w" );
         if( f )     // able to write
         {
             std::fclose( f );
-            std::remove( filename.toAscii() );
+            std::remove( filename.toLatin1() );
```
Comment (StasJ) [inline apps/vaporgui/FileOperationChecker.cpp]: Why do you use both `toLocal8Bit` and `toLatin1` for the same thing?

File: apps/vaporgui/RenderEventRouter.cpp [hunk 7]
```diff
@@ -88,9 +88,6 @@ string RenderEventRouter::GetIconImagePath() const {
 
 
 void RenderEventRouter::setEditorDirty() {
-#ifndef WIN32
-	#warning What is this? I had to delete the code here and it didn't match the doxygen
```
Comment (StasJ) [inline apps/vaporgui/RenderEventRouter.cpp]: Why was this removed and the empty function kept?

File: apps/vaporgui/SettingsParams.cpp [hunk 8]
```diff
@@ -84,7 +84,7 @@ SettingsParams::SettingsParams(
 ) : ParamsBase(ssave, _classType) {
 
 	_settingsPath = QDir::homePath().toStdString();
-	_settingsPath += QDir::separator().toAscii();
+    _settingsPath += "/";
```
Comment (StasJ) [inline apps/vaporgui/SettingsParams.cpp]: Does this separator work in Windows?

File: apps/vaporgui/TFOpacityWidget.cpp [hunk 9]
```diff
@@ -130,7 +130,7 @@ void TFOpacityMap::paintEvent(QPainter &p)
 void TFOpacityMap::mousePressEvent(QMouseEvent *event)
 {
     emit Activated(this);
-    vec2 mouse = qvec2(event->posF());
+	vec2 mouse = vec2(1.0, 1.0);//qvec2(event->posF());
```
Comment (StasJ) [inline apps/vaporgui/TFOpacityWidget.cpp]: Uhh I see here why you were saying the OpacityMap didn't work.

File: apps/vaporgui/VSection.cpp [hunk 10]
```diff
@@ -46,7 +46,7 @@ QString VSection::_createStylesheet() const
     }
     )";
 #elif defined(WIN32)
-#error style missing for windows
+//#error style missing for windows
```
Comment (StasJ) [inline apps/vaporgui/VSection.cpp]: Please un-delete this error

File: apps/vaporgui/gui.pro [hunk 11]
```diff
@@ -31,13 +31,14 @@ win32:QMAKE_CXXFLAGS_DEBUG += /EHsc
...
 
+QT += widgets
 
 INCLUDEPATH += . \
 	../../include \
 	../../../apps/vaporgui/misc \
 	../../../apps/vaporgui/guis \
 	../../../apps/vaporgui/main \
-	../../../apps/vaporgui/render
+	../../../apps/vaporgui/render \
```
Comment (StasJ) [inline apps/vaporgui/gui.pro]: What are these pointing to?

File: apps/vaporgui/main.cpp [hunk 12]
```diff
@@ -71,9 +71,7 @@ FILE *OpenLog(string path_var) {
 	return(fp);
 }
 
-#ifndef WIN32
-#warning Qt4 uses deprecated OSX calls which pollute the console with warnings
-#endif
+
```
Comment (StasJ) [inline apps/vaporgui/main.cpp]: Please undelete this warning.

File: site_files/site.NCAR [hunk 13]
```diff
@@ -1,61 +1,45 @@
...
-elseif (WIN32 AND NOT EXISTS "${WINDOWS_THIRD_PARTY_DIR}")
-	return ()
-endif ()
-
-
 if (APPLE)
-	set (THIRD_PARTY_DIR "${APPLE_THIRD_PARTY_DIR}")
+    set (THIRD_PARTY_DIR     /usr/local/VAPOR-Deps/2019-Aug			     )
+    set (QTDIR               /usr/local/VAPOR-Deps/2019-Aug/Qt/5.12.4/clang_64       )
+    set (CMAKE_PREFIX_PATH   ${QTDIR}/lib/cmake                                      )
```
Comment (StasJ) [inline site_files/site.NCAR]: You are clobbering anything the user might place in CMAKE_PREFIX_PATH

File: site_files/site.NCAR [hunk 14]
```diff
@@ -1,61 +1,45 @@
...
+    set (Qt5Core_DIR          ${QTDIR}/lib/cmake/Qt5Core )
+    set (QT_QMAKE_EXECUTABLE  ${QTDIR}/gcc_64/bin/qmake )
 else ()
-	set (THIRD_PARTY_DIR "${WINDOWS_THIRD_PARTY_DIR}")
+	set (THIRD_PARTY_DIR    C:/2019-Aug)
+    set (QTDIR              ${THIRD_PARTY_DIR}/Qt5_12_4/5.12.4/msvc2015_64)
+    set (CMAKE_PREFIX_PATH  ${QTDIR}/lib/cmake/Qt5) 
 endif ()
 
+find_package(Qt5 COMPONENTS Core Widgets REQUIRED)
```
Comment (StasJ) [inline site_files/site.NCAR]: This should not be done in the site file.

File: apps/vaporgui/TFOpacityWidget.cpp (hunk 9 above)
Comment (StasJ) [reply apps/vaporgui/TFOpacityWidget.cpp]: #1934

File: apps/vaporgui/main.cpp (hunk 12 above)
Comment (StasJ) [reply apps/vaporgui/main.cpp]: Never mind, but please remove the code that this warning warns about.

File: apps/vaporgui/CMakeLists.txt (hunk 5 above)
Comment (sgpearse) [reply apps/vaporgui/CMakeLists.txt]: Done.  "assimp-vc140-mt" is the name of the library created in VS2014.  I think we should stick to default options when building the Win32 libraries for simplicity.

File: apps/vaporgui/RenderEventRouter.cpp (hunk 7 above)
Comment (sgpearse) [reply apps/vaporgui/RenderEventRouter.cpp]: I see your point.  The function is removed now.

File: apps/vaporgui/SettingsParams.cpp (hunk 8 above)
Comment (sgpearse) [reply apps/vaporgui/SettingsParams.cpp]: It does, but I agree that this can be improved.  Done.

File: apps/vaporgui/TFOpacityWidget.cpp (hunk 9 above)
Comment (sgpearse) [reply apps/vaporgui/TFOpacityWidget.cpp]: My mistake.  Sorry about this.

File: apps/vaporgui/VSection.cpp (hunk 10 above)
Comment (sgpearse) [reply apps/vaporgui/VSection.cpp]: I've removed the #elif for WIN32, so it's using the stylesheet for Linux.  Shown below:

unnamed

File: apps/vaporgui/gui.pro (hunk 11 above)
Comment (sgpearse) [reply apps/vaporgui/gui.pro]: This must be legacy code that has slipped through the cracks.  Removed.

File: apps/vaporgui/main.cpp (hunk 12 above)
Comment (sgpearse) [reply apps/vaporgui/main.cpp]: Since this branch is on Qt5, I think we can remove the warning entirely, no?

File: site_files/site.NCAR (hunk 13 above)
Comment (sgpearse) [reply site_files/site.NCAR]: Good catch.

File: site_files/site.NCAR (hunk 14 above)
Comment (sgpearse) [reply site_files/site.NCAR]: Good catch, thank you.

File: apps/vaporgui/VSection.cpp (hunk 10 above)
Comment (StasJ) [reply apps/vaporgui/VSection.cpp]: Great, glad it works, thanks!

File: apps/vaporgui/main.cpp (hunk 12 above)
Comment (StasJ) [reply apps/vaporgui/main.cpp]: Yes, but this warning warns about code to hide some deprecated warnings from Qt4 so we need to remove the associated code too.

File: apps/vaporgui/MainForm.cpp [hunk 15]
```diff
@@ -1267,11 +1267,11 @@ void MainForm::_fileSaveHelper(string path)
 
         QFileDialog fileDialog(this, "Save VAPOR session file",
                     QString::fromStdString( dir ),
-                    QString::fromAscii("Vapor 3 Session Save file (*.vs3)" )
+                    QString("Vapor 3 Session Save file (*.vs3)" )
```
Comment (shaomeng) [inline apps/vaporgui/MainForm.cpp]:
```suggestion
                    QString::fromLatin1("Vapor 3 Session Save file (*.vs3)" )
```

File: apps/vaporgui/MainForm.cpp [hunk 16]
```diff
@@ -1267,11 +1267,11 @@ void MainForm::_fileSaveHelper(string path)
...
         );
 
         fileDialog.setAcceptMode(QFileDialog::AcceptSave);
-        fileDialog.setDefaultSuffix( QString::fromAscii("vs3") );
+        fileDialog.setDefaultSuffix( QString("vs3") );
```
Comment (shaomeng) [inline apps/vaporgui/MainForm.cpp]:
```suggestion
        fileDialog.setDefaultSuffix( QString::fromLatin1("vs3") );
```

File: apps/vaporgui/FileOperationChecker.cpp [hunk 17]
```diff
@@ -92,11 +92,11 @@ FileOperationChecker::FileGoodToWrite( const QString& filename )
     // In case this file does not exist
     if( !fileInfo.exists() )
     {
-        std::FILE* f = std::fopen( filename.toAscii(), "w" );
+        std::FILE* f = std::fopen( filename.toLocal8Bit(), "w" );
```
Comment (shaomeng) [inline apps/vaporgui/FileOperationChecker.cpp]:
```suggestion
        std::FILE* f = std::fopen( filename.toLatin1(), "w" );
```

File: apps/vaporgui/VaporTable.cpp [hunk 18]
```diff
@@ -125,7 +128,8 @@ bool VaporTable::GetAutoResizeHeight() const {
 void VaporTable::StretchToColumn(int column) {
     QHeaderView *headerView = new QHeaderView(Qt::Horizontal, _table);
     _table->setHorizontalHeader(headerView);
-    headerView->setResizeMode(column, QHeaderView::Stretch);
+    //headerView->setResizeMode(column, QHeaderView::Stretch);
```
Comment (shaomeng) [inline apps/vaporgui/VaporTable.cpp]:
```suggestion
```

File: buildutils/OSX_PostBuild.py [hunk 19]
```diff
@@ -0,0 +1,79 @@
+# On OSX, ld builds libraries that contain hard coded path to their dependencies.
+# After installation on a client machiene, these paths become invalid unless we 
+# modify them to reference @rpath.  This script adds @rpath!
+
```
Comment (shaomeng) [inline buildutils/OSX_PostBuild.py]: I wonder what mechanism we used before to tackle this problem?

File: lib/common/ResourcePath.cpp [hunk 20]
```diff
@@ -1,3 +1,4 @@
+#include <iostream>
```
Comment (shaomeng) [inline lib/common/ResourcePath.cpp]:
```suggestion
```

File: lib/common/ResourcePath.cpp [hunk 21]
```diff
@@ -97,13 +98,16 @@ std::string Wasp::GetSharePath(const std::string &name)
...
 
-
 #ifdef WIN32
 #define PYTHON_INSTALLED_PATH ("python" + string(PYTHON_VERSION))
 #else
 #define PYTHON_INSTALLED_PATH ("lib/python" + string(PYTHON_VERSION))
 #endif
 
+std::string Wasp::GetPythonVersion() {
+	return string(PYTHON_VERSION);
```
Comment (shaomeng) [inline lib/common/ResourcePath.cpp]:
```suggestion
	return std::string(PYTHON_VERSION);
```

File: lib/vdc/DerivedVar.cpp [hunk 22]
```diff
@@ -361,7 +361,7 @@ void resampleToStaggered(
 		inDims.push_back(inMax[i]-inMin[i]+1);
 		outDims.push_back(outMax[i]-outMin[i]+1);
 	}
-	size_t sz = std::max(vproduct(outDims), vproduct(inDims));
+	size_t sz = (std::max)(vproduct(outDims), vproduct(inDims));
```
Comment (shaomeng) [inline lib/vdc/DerivedVar.cpp]: This syntax looks weird. What problem does it fix?

Comment (shaomeng) [reply lib/vdc/DerivedVar.cpp]: The same comment applies to the rest changes in this file.

File: site_files/site.NCAR [hunk 23]
```diff
@@ -1,61 +1,44 @@
...
+	set (THIRD_PARTY_DIR    C:/2019-Aug)
+    set (QTDIR              ${THIRD_PARTY_DIR}/Qt5_12_4/5.12.4/msvc2015_64)
+    set (CMAKE_PREFIX_PATH  ${QTDIR}/lib/cmake/Qt5) 
 endif ()
 
 # Yellowstone and Geyser configuration
 # --------------------------------------------------------------------------------
 cmake_host_system_information(RESULT HOSTNAME QUERY HOSTNAME)
-if (${ON_GEYSER_OR_CALDERA})
+if (HOSTNAME MATCHES "yslogin[0-9]+" OR HOSTNAME MATCHES "geyser[0-9]+")
```
Comment (shaomeng) [inline site_files/site.NCAR]: Shall we delete all references to yellowstone and geyser?

File: site_files/site.NCAR [hunk 24]
```diff
@@ -1,61 +1,44 @@
...
+set (THIRD_PARTY_LIB_DIR "${THIRD_PARTY_DIR}/lib")
+
 if (UNIX)
-	set (THIRD_PARTY_LIB_DIR "${THIRD_PARTY_DIR}/lib")
-	set (QTDIR ${THIRD_PARTY_DIR})
-	set (PYTHONVERSION 2.7)
+	set (PYTHONVERSION 3.6)
 	set (PYTHONDIR ${THIRD_PARTY_DIR})
-	set (PYTHONPATH ${THIRD_PARTY_LIB_DIR}/python${PYTHONVERSION})
+	set (PYTHONPATH ${THIRD_PARTY_LIB_DIR}/python3.6)
```
Comment (shaomeng) [inline site_files/site.NCAR]: Is it a good idea to hand code versions?

File: CMakeLists.txt [hunk 25]
```diff
@@ -94,27 +94,43 @@ include_directories ("${PROJECT_SOURCE_DIR}/include")
...
+
 if (WIN32)
-	include_directories (${PYTHONPATH}/Lib/site-packages/numpy/core/include)
+	include_directories (${THIRD_PARTY_DIR}/Python${PYTHONVERSION}/include)
+	include_directories (${NUMPY_INCLUDE_DIR})
+	get_property(dirs DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY INCLUDE_DIRECTORIES)
 else ()
-	include_directories (${PYTHONPATH}/site-packages/numpy/core/include)
+	#include_directories (${THIRD_PARTY_INC_DIR}/python${PYTHONVERSION})
+	include_directories (${THIRD_PARTY_INC_DIR}/python3.6m)
```
[the remark this answers is not in the document]
Comment (sgpearse) [reply CMakeLists.txt]: You are correct.  Fixed.

File: CMakeLists.txt [hunk 26]
```diff
@@ -94,27 +94,43 @@ include_directories ("${PROJECT_SOURCE_DIR}/include")
...
+    find_library(TIFF libtiff)
+    find_library(PROJ proj_6_1)
+else ()
+	find_library(ASSIMP assimp)
+    find_library(TIFF tiff)
+    find_library(PROJ proj)
+endif()
 
 if (BUILD_VDC)
 	find_package (OpenGL REQUIRED)
```
[the remark this answers is not in the document]
Comment (sgpearse) [reply CMakeLists.txt]: Not sure.  I think we should revisit it since it's outside of the scope of this PR.

File: CMakeLists.txt [hunk 27]
```diff
@@ -195,18 +215,28 @@ if (WIN32)
...
+		set (DLL_DIR ${THIRD_PARTY_DIR}/dll)
 		file (GLOB WIN_INSTALL_DLLS ${DLL_DIR}/*.dll ${QTDIR}/bin/*.dll)
+
 		install (
 			FILES ${WIN_INSTALL_DLLS}
 			DESTINATION ${INSTALL_BIN_DIR}
 			COMPONENT Dependencies
 			)
+		install (
+			FILES ${PYTHONPATH}/python36.dll
```
[the remark this answers is not in the document]
Comment (sgpearse) [reply CMakeLists.txt]: Fixed.

File: CMakeLists.txt [hunk 28]
```diff
@@ -266,16 +310,40 @@ if (UNIX AND NOT APPLE)
...
+
 		install (
 			FILES ${INSTALL_LIBS}
 			DESTINATION ${INSTALL_LIB_DIR}
 			COMPONENT Dependencies
 			)
+
+		install (
+			#DIRECTORY ${THIRD_PARTY_LIB_DIR}/python${PYTHONVERSION}
+			DIRECTORY ${THIRD_PARTY_LIB_DIR}/python3.6
```
[the remark this answers is not in the document]
Comment (sgpearse) [reply CMakeLists.txt]: Fixed.  Thank you.

File: apps/tiff2geotiff/tiff2geotiff.cpp [hunk 29]
```diff
@@ -18,6 +18,8 @@
 #include <ctype.h>
 #include "vapor/VAssert.h"
 
+#define ACCEPT_USE_OF_DEPRECATED_PROJ_API_H 1
```
[the remark this answers is not in the document]
Comment (sgpearse) [reply apps/tiff2geotiff/tiff2geotiff.cpp]: I think there are good reasons to do this, but I'm not certain if this needs to be done so close to release.

File: apps/tiff2geotiff/tiff2geotiff.cpp [hunk 30]
```diff
@@ -428,7 +430,7 @@ static int applyCorners(float lonlat[4], float relPos[4], TIFF* out){
 		//reproject latlon to specified coord projection.
 		//unless it's already a lat/lon projection
 		double dbextents[4];
-		if ( pj_is_latlong(p)) {
+		if ( pj_is_latlong((projPJ)p)) {
```
[the remark this answers is not in the document]
Comment (sgpearse) [reply apps/tiff2geotiff/tiff2geotiff.cpp]: Good catch.  Fixed.

File: apps/vaporgui/FileOperationChecker.cpp (hunk 17 above)
Comment (sgpearse) [reply apps/vaporgui/FileOperationChecker.cpp]: Fixed.

File: apps/vaporgui/FlowSubtabs.cpp [hunk 31]
```diff
@@ -152,12 +152,12 @@ FlowSeedingSubtab::FlowSeedingSubtab(QWidget* parent) : QVaporSubtab(parent)
 
     /* The following two widgets deal with flow line output and seed point input */
     _fileWriter = new VFileWriter( this, "Output Flow Lines" );
-    _fileWriter->SetFileFilter( QString::fromAscii("*.txt") );
+    _fileWriter->SetFileFilter( QString("*.txt") );
```
[the remark this answers is not in the document]
Comment (sgpearse) [reply apps/vaporgui/FlowSubtabs.cpp]: Do you think that the default constructor for QString is unsuitable in this case?

File: apps/vaporgui/VaporTable.cpp (hunk 18 above)
Comment (sgpearse) [reply apps/vaporgui/VaporTable.cpp]: Fixed.

File: buildutils/OSX_PostBuild.py (hunk 19 above)
Comment (sgpearse) [reply buildutils/OSX_PostBuild.py]: Me too.

File: lib/common/ResourcePath.cpp (hunk 20 above)
Comment (sgpearse) [reply lib/common/ResourcePath.cpp]: Fixed.

File: lib/common/ResourcePath.cpp (hunk 21 above)
Comment (sgpearse) [reply lib/common/ResourcePath.cpp]: Fixed.

File: lib/vdc/DerivedVar.cpp (hunk 22 above)
Comment (sgpearse) [reply lib/vdc/DerivedVar.cpp]: Windows.h has a macro that overrides the max token, std::max will evaluate into the expansion of the macro.  Putting it in parenthesis prevents the compiler from inserting the macro.

File: site_files/site.NCAR (hunk 23 above)
Comment (sgpearse) [reply site_files/site.NCAR]: I don't see why not, but I'd like to change as few things as possible and keep make this PR focused on 3rd party libraries.

File: site_files/site.NCAR (hunk 24 above)
Comment (sgpearse) [reply site_files/site.NCAR]: Nope.  Thank you.  Fixed.

File: CMakeLists.txt (hunk 26 above)
[the remark this answers is not in the document]
Comment (sgpearse) [reply CMakeLists.txt]: @clyne would you mind weighing in on this before we resolve the issue?

File: lib/vdc/DerivedVar.cpp (hunk 22 above)
Comment (StasJ) [reply lib/vdc/DerivedVar.cpp]: You could undefine the macro

File: CMakeLists.txt (hunk 26 above)
[the remark this answers is not in the document]
Comment (StasJ) [reply CMakeLists.txt]: Originally, there was OpenGL code in the VDC library. I think it was removed.

[the remark this answers is not in the document]
Comment (clyne) [reply CMakeLists.txt]: There *should* not be any OpenGL dependencies in the VDC library. I 'd try to remove this.
